### PR TITLE
feat: depend on marl-eval for json logger

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
We moved the json logger to `marl-eval` this PR makes Mava depend on `marl-eval` for logging json files. 

## Note 
This PR can only be merged once this [PR](https://github.com/instadeepai/marl-eval/pull/46) has been merged on marl-eval.

### CODE REVIEW

File: mava/utils/logger.py [hunk 1]
```diff
@@ -340,101 +342,4 @@ def describe(x: ArrayLike) -> Union[Dict[str, ArrayLike], ArrayLike]:
...
-            } | metrics
-            step_str = f"step_{evaluation_step}"
-            if step_str in self.run_data:
-                self.run_data[step_str].update(step_metrics)
-            else:
-                self.run_data[step_str] = step_metrics
-
-        with open(self.file_path, "w") as f:
-            json.dump(self.data, f, indent=4)
+    return {"mean": np.mean(x), "std": np.std(x), "min": np.min(x), "max": np.max(x)}
```
Comment (sash-a) [inline mava/utils/logger.py]: repeated

```suggestion
```

File: mava/utils/logger.py [hunk 2]
```diff
@@ -13,20 +13,19 @@
...
 from enum import Enum
-from typing import Dict, List, Optional, Union
+from typing import Dict, List, Union
 
 import jax
 import neptune
 import numpy as np
 from colorama import Fore, Style
 from jax.typing import ArrayLike
+from marl_eval.json_tools import JsonWriter
```
Comment (sash-a) [inline mava/utils/logger.py]: Then I'd say we should rename the mava `JsonLogger` to `MarlEvalLogger`

```suggestion
from marl_eval.json_tools import JsonLogger
```

**Review by sash-a — CHANGES_REQUESTED** [review]
Looking good, just a recommendation because of the name change in the marl-eval PR :fire:

File: mava/utils/logger.py (hunk 2 above)
Comment (RuanJohn) [reply mava/utils/logger.py]: Done :)

File: mava/utils/logger.py (hunk 1 above)
Comment (RuanJohn) [reply mava/utils/logger.py]: Good catch. :raised_hands:

File: mava/utils/logger.py [hunk 3]
```diff
@@ -222,7 +221,7 @@ def __init__(self, cfg: DictConfig, unique_token: str) -> None:
                 cfg.logger.base_exp_path, "json", cfg.logger.kwargs.json_path
             )
 
-        self.logger = JsonWriter(
+        self.logger = MarlEvalJsonLogger(
```
Comment (WiemKhlifi) [inline mava/utils/logger.py:224]: Cool 😍

**Review by sash-a — APPROVED** [review]
Baie lekker! :balloon:

**Review by WiemKhlifi — APPROVED** [review]
Thanks @RuanJohn 🔥 Works great 🚀

**Pull request merged**